### PR TITLE
Changes in test/ resulting from the audit

### PR DESCRIPTION
**Pull request by IaroslavMazur**

### CODE REVIEW

**Comment by IaroslavMazur** [comment]
@PaulRBerg, see [this comment](https://github.com/sablier-labs/v2-periphery/pull/139#issuecomment-1627688682) that also relates to this PR

File: test/integration/target/burn/burn.t.sol [hunk 1]
```diff
@@ -44,7 +44,7 @@ contract Burn_Integration_Test is Integration_Test {
         data = abi.encodeCall(target.burn, (lockup, streamId));
         aliceProxy.execute(address(target), data);
 
-        // Expect the NFT owner not to exist anymore.
+        // Expect the NFT owner to not exist anymore.
```
Comment (PaulRBerg) [inline test/integration/target/burn/burn.t.sol:47]: 😄

File: test/integration/target/cancel-multiple/cancelMultiple.t.sol [hunk 2]
```diff
@@ -23,15 +23,15 @@ contract CancelMultiple_Integration_Test is Integration_Test {
...
     }
 
     function test_CancelMultiple_LockupLinear() external whenDelegateCalled {
         uint256[] memory streamIds = batchCreateWithRange();
-        test_CancelMultiple(lockupLinear, streamIds);
+        cancelMultiple(lockupLinear, streamIds);
     }
 
-    function test_CancelMultiple(ISablierV2Lockup lockup, uint256[] memory streamIds) internal {
+    function cancelMultiple(ISablierV2Lockup lockup, uint256[] memory streamIds) internal {
```
Comment (PaulRBerg) [inline test/integration/target/cancel-multiple/cancelMultiple.t.sol:34]: `test_cancelMultiple` is ok

File: test/integration/target/renounce/renounce.t.sol [hunk 3]
```diff
@@ -19,15 +19,15 @@ contract Renounce_Integration_Test is Integration_Test {
...
     }
 
     function test_Renounce_LockupLinear() external whenDelegateCalled {
         uint256 streamId = createWithRange();
-        test_Renounce(lockupLinear, streamId);
+        renounce(lockupLinear, streamId);
     }
 
-    function test_Renounce(ISablierV2Lockup lockup, uint256 streamId) internal {
+    function renounce(ISablierV2Lockup lockup, uint256 streamId) internal {
```
Comment (PaulRBerg) [inline test/integration/target/renounce/renounce.t.sol:30]: `test_Renounce` is ok

File: test/integration/target/withdraw-max-and-transfer/withdrawMaxAndTransfer.t.sol [hunk 4]
```diff
@@ -22,17 +22,17 @@ contract WithdrawMaxAndTransfer_Integration_Test is Integration_Test {
...
     function test_WithdrawMaxAndTransfer_LockupLinear() external whenDelegateCalled {
         LockupLinear.CreateWithRange memory params = defaults.createWithRange();
         params.recipient = address(aliceProxy);
         uint256 streamId = createWithRange(params);
-        test_WithdrawMaxAndTransfer(lockupLinear, streamId);
+        withdrawMaxAndTransfer(lockupLinear, streamId);
     }
 
-    function test_WithdrawMaxAndTransfer(ISablierV2Lockup lockup, uint256 streamId) internal {
+    function withdrawMaxAndTransfer(ISablierV2Lockup lockup, uint256 streamId) internal {
```
Comment (PaulRBerg) [inline test/integration/target/withdraw-max-and-transfer/withdrawMaxAndTransfer.t.sol:35]: `test_WithdrawMaxAndTransfer` is ok

File: test/fork/plugin/onStreamCanceled.t.sol [hunk 5]
```diff
@@ -9,7 +9,7 @@ import { Fork_Test } from "../Fork.t.sol";
 
 import { Permit2Params } from "src/types/Permit2.sol";
 
-/// @dev Runs against multiple assets.
+/// @dev Runs against the asset passed to its constructor.
```
Comment (PaulRBerg) [inline test/fork/plugin/onStreamCanceled.t.sol:12]: Yes, and there are multiple instances of this contract. I'll rewrite the comment to say this:

> Runs against multiple fork assets

File: test/Base.t.sol [hunk 6]
```diff
@@ -428,14 +428,22 @@ abstract contract Base_Test is Assertions, Events, StdCheats, V2CoreUtils {
...
-        return abi.decode(response, (uint256));
+        return decodeAsUint256(response);
+    }
+
+    function decodeAsUint256(bytes memory input) private pure returns (uint256) {
+        return abi.decode(input, (uint256));
+    }
+
+    function decodeAsUint256Array(bytes memory input) private pure returns (uint256[] memory) {
+        return abi.decode(input, (uint256[]));
```
Comment (PaulRBerg) [inline test/Base.t.sol:447]: Adding these functions increases the number of lines of code - let's apply the principle of KISS and remove them

**Review by PaulRBerg — CHANGES_REQUESTED** [review]
Thanks, @IaroslavMazur.

**Review by PaulRBerg — APPROVED** [review]
LGTM now (after my recent commit)

**Comment by PaulRBerg** [comment]
Merging - fork tests are failing because @IaroslavMazur's fork does not have an `RPC_URL_MAINNET` defined in the GitHub Secrets.

File: test/integration/target/burn/burn.t.sol [hunk 7]
```diff
@@ -22,17 +22,17 @@ contract Burn_Integration_Test is Integration_Test {
...
     function test_Burn_LockupLinear() external whenDelegateCalled {
         LockupLinear.CreateWithRange memory params = defaults.createWithRange();
         params.recipient = address(aliceProxy);
         uint256 streamId = createWithRange(params);
-        test_Burn(lockupLinear, streamId);
+        burn(lockupLinear, streamId);
     }
 
-    function test_Burn(ISablierV2Lockup lockup, uint256 streamId) internal {
+    function burn(ISablierV2Lockup lockup, uint256 streamId) internal {
```
[the remark this answers is not in the document]
Comment (IaroslavMazur) [reply test/integration/target/burn/burn.t.sol]: Hm... Fair point.

Though, that also speaks to my original concern with the naming: `test_Burn` looks too much like a test that's supposed to be executed by Foundry. The only thing that differentiates it from the ones that do get executed is the `internal` modifier. 😕

File: test/utils/BatchBuilder.sol [hunk 8]
```diff
@@ -6,9 +6,9 @@ import { LockupDynamic, LockupLinear } from "@sablier/v2-core/types/DataTypes.so
 import { Batch } from "../../src/types/DataTypes.sol";
 
 library BatchBuilder {
-    /// @notice Generates an array of `batchSingle` with the specified `batchSize`.
-    function fillBatch(
-        Batch.CreateWithDeltas memory batchSingle,
+    /// @notice Generates an array containing `batchSize` copies of `sample`.
+    function generateBatchFromSample(
```
[the remark this answers is not in the document]
Comment (IaroslavMazur) [reply test/utils/BatchBuilder.sol]: Oh... And I was so proud for having come with the whole "sample to batch" framework... 😢

File: test/integration/target/burn/burn.t.sol (hunk 7 above)
[the remark this answers is not in the document]
Comment (PaulRBerg) [reply test/integration/target/burn/burn.t.sol]: Acknowledged

**Pull request merged**